plugin/cql/src/main/java/org/opencds/cqf/ruler/cql/r4/ArtifactAssessment.java

```diff
@@ -0,0 +1,1336 @@
+			if (codeString == null || "".equals(codeString))
+				return null;
+		if ("unresolved".equals(codeString))
+			return new Enumeration<ArtifactAssessmentDisposition>(this, ArtifactAssessmentDisposition.UNRESOLVED);
+		if ("not-persuasive".equals(codeString))
+			return new Enumeration<ArtifactAssessmentDisposition>(this, ArtifactAssessmentDisposition.NOTPERSUASIVE);
+		if ("persuasive".equals(codeString))
+			return new Enumeration<ArtifactAssessmentDisposition>(this, ArtifactAssessmentDisposition.PERSUASIVE);
+		if ("persuasive-with-modification".equals(codeString))
+			return new Enumeration<ArtifactAssessmentDisposition>(this, ArtifactAssessmentDisposition.PERSUASIVEWITHMODIFICATION);
+		if ("not-persuasive-with-modification".equals(codeString))
+			return new Enumeration<ArtifactAssessmentDisposition>(this, ArtifactAssessmentDisposition.NOTPERSUASIVEWITHMODIFICATION);
+		throw new FHIRException("Unknown ArtifactAssessmentDisposition code '"+codeString+"'");
+		}
+public String toCode(ArtifactAssessmentDisposition code) {
+	if (code == ArtifactAssessmentDisposition.UNRESOLVED)
+		return "unresolved";
+	if (code == ArtifactAssessmentDisposition.NOTPERSUASIVE)
+		return "not-persuasive";
+	if (code == ArtifactAssessmentDisposition.PERSUASIVE)
+		return "persuasive";
+	if (code == ArtifactAssessmentDisposition.PERSUASIVEWITHMODIFICATION)
+		return "persuasive-with-modification";
+	if (code == ArtifactAssessmentDisposition.NOTPERSUASIVEWITHMODIFICATION)
+		return "not-persuasive-with-modification";
+	return "?";
+	}
+public String toSystem(ArtifactAssessmentDisposition code) {
+	return code.getSystem();
+	}
+}
+
+	public static final String ARTIFACT_COMMENT_EXTENSION_URL = "http://hl7.org/fhir/us/cqfmeasures/StructureDefinition/cqfm-artifactComment";
+	public static final String CONTENT = "http://hl7.org/fhir/uv/crmi/StructureDefinition/crmi-artifactAssessmentContent";
+	public static final String ARTIFACT = "http://hl7.org/fhir/uv/crmi/StructureDefinition/crmi-artifactAssessmentArtifact";
+	public static final String CITEAS = "http://hl7.org/fhir/uv/crmi/StructureDefinition/crmi-artifactAssessmentCiteAs";
+	public static final String TITLE = "http://hl7.org/fhir/uv/crmi/StructureDefinition/crmi-artifactAssessmentTitle";
+	public static final String DATE = "http://hl7.org/fhir/uv/crmi/StructureDefinition/crmi-artifactAssessmentDate";
+	public static final String COPYRIGHT = "http://hl7.org/fhir/uv/crmi/StructureDefinition/crmi-artifactAssessmentCopyright";
+	public static final String APPROVAL_DATE = "http://hl7.org/fhir/uv/crmi/StructureDefinition/crmi-artifactAssessmentApprovalDate";
+	public static final String LAST_REVIEW_DATE = "http://hl7.org/fhir/uv/crmi/StructureDefinition/crmi-artifactAssessmentLastReviewDate";
+	public static final String WORKFLOW_STATUS = "http://hl7.org/fhir/uv/crmi/StructureDefinition/crmi-artifactAssessmentWorkflowStatus";
+	public ArtifactAssessment(){
+		super();
+	}
```

Comment:
ArtifactAssessment.artifact is a required field, I wanted to omit this constructor to avoid creating invalid instances but I kept getting compiler errors, is there a way to remove it? Or to make it clear that the other constructors are preferred?